apache/superset
# fix(Dashboard): Retain colors when color scheme not set

### SUMMARY
This PR does the following:
- It fixes an issue for which colors would sometimes not keep their consistency when leaving and revisiting the Dashboard.
- It fixes an issue for which charts original color schemes were not tracked making falling back to their origin color scheme impossible when removing the color scheme from the Dashboard.
- It attempts to fix and re-enable skipped E2E tests that are critical for the coverage of the color consistency implementation.
- Fixes https://github.com/apache/superset/issues/23874 with limitations:
  - The color scheme cannot be changed if the chart shares labels with multiple charts in the Dashboard. 
  - When setting custom label colors via the advanced settings of the Dashboard, the custom label colors will always be enforced also when a color scheme is not set for the Dashboard. 
  - The chart should consistently display colors that match those used in the Dashboard. 

### BEFORE

https://github.com/user-attachments/assets/d66e66ff-13ec-4bb1-a1bf-72f6afcd1bfc

### AFTER

https://github.com/user-attachments/assets/fd74e76e-ea20-4d89-8f05-49455bcb5fd0


### TESTING INSTRUCTIONS
1. Open a Dashboard with a color scheme and shared label colors
2. Edit the Dashboard properties and remove the color scheme
3. After applying the changes, verify that shared label colors are still respected
4. For charts that do not share labels, verify their original color scheme is respected
5. Save and exit the Dashboard by going to the Dashboard lists
6. Come back again to the same Dashboard you just saved
7. Make sure colors haven't changed from the previous state

### ADDITIONAL INFORMATION
